### PR TITLE
Enable TLS in integration tests

### LINKED ISSUE #266: Node to node encryption
As a user, I need all my node traffic to be sent encrypted in order to safeguard my privacy.

Currently, we have a single way to receive / exchange data: HTTP REST queries.
In the future, we want to move to a binary protocol, but should still be using SSL to transmit data.
Some method might still be exposed via REST and it should be over HTTPS.

### PR DESCRIPTION
**Pull request by omerfirmak**
Fixes #266

### CODE REVIEW

**Comment by omerfirmak** [comment]
Needs https://github.com/bosagora/faucet/pull/108

File: ci/system_integration_test.d [hunk 1]
```diff
@@ -65,6 +78,15 @@ private int main (string[] args)
...
 
+    // Generate TLS files
+    foreach (dir; NodeDirs)
+    {
+        immutable key_file = dir ~ "agora-key.pem";
+        immutable cert_file = dir ~ "agora-cert.pem";
+        runCmd(["rm", "-f", cert_file, key_file]);
+        runCmd(GenCerts ~ ["-keyout", key_file, "-out", cert_file,]);
+    }
+
```
Comment (Geod24) [inline ci/system_integration_test.d]: Why not just generate it and `git add` it ?
This way, one can run the integration test manually (`docker compose up`) without using this file.

**Comment by Geod24** [comment]
```
node-7_1  | 2021-06-17T08:23:18.672545870Z SSL modified options: passed 0x00020000 vs applied 0x00000000
node-7_1  | 2021-06-17T08:23:18.682330813Z SSL modified options: passed 0x00020000 vs applied 0x00000000
node-7_1  | 2021-06-17T08:23:18.682417019Z SSL modified options: passed 0x00020000 vs applied 0x00000000
node-7_1  | 2021-06-17T08:23:18.691026373Z SSL modified options: passed 0x00020000 vs applied 0x00000000
node-7_1  | 2021-06-17T08:23:18.691629019Z SSL modified options: passed 0x00020000 vs applied 0x00000000
node-7_1  | 2021-06-17T08:23:18.706294232Z SSL modified options: passed 0x00020000 vs applied 0x00000000
```

File: tests/system/dub.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "targetType": "executable",
     "targetPath": "build",
     "dflags": [ "-i", "-preview=in", "-revert=dtorfields" ],
-    "libs-posix": [ "sodium", "sqlite3" ],
+    "libs-posix": [ "sodium", "sqlite3", "openssl" ],
```
Comment (Geod24) [inline tests/system/dub.json:6]: I don't think you need this, do you ?
The Vibe.d package [depends on `vibe-d:tls` by default](https://github.com/vibe-d/vibe.d/blob/d638b343d6f9ac8da5b18dff61c772d92aa31aa3/dub.sdl#L20) which itself depends on [openssl, including this lib statement](https://github.com/vibe-d/vibe.d/blob/d638b343d6f9ac8da5b18dff61c772d92aa31aa3/tls/dub.sdl#L22-L84).

**Comment by omerfirmak** [comment]
> ```
> node-7_1  | 2021-06-17T08:23:18.672545870Z SSL modified options: passed 0x00020000 vs applied 0x00000000
> node-7_1  | 2021-06-17T08:23:18.682330813Z SSL modified options: passed 0x00020000 vs applied 0x00000000
> node-7_1  | 2021-06-17T08:23:18.682417019Z SSL modified options: passed 0x00020000 vs applied 0x00000000
> node-7_1  | 2021-06-17T08:23:18.691026373Z SSL modified options: passed 0x00020000 vs applied 0x00000000
> node-7_1  | 2021-06-17T08:23:18.691629019Z SSL modified options: passed 0x00020000 vs applied 0x00000000
> node-7_1  | 2021-06-17T08:23:18.706294232Z SSL modified options: passed 0x00020000 vs applied 0x00000000
> ```

Those seemed to be harmless for the most part.

File: tests/system/dub.json (hunk 2 above)
Comment (omerfirmak) [reply tests/system/dub.json:6]: I too expected it to work that way, but it didnt for some reason. I got linking errors without explicit `openssl` in the libraries.

Comment (Geod24) [reply tests/system/dub.json:6]: I guess we're at that stage where "Because dub" is a good enough explanation for any change.

**Comment by omerfirmak** [comment]
I dont get why it is failing to complete discovery now tho.

**Comment by omerfirmak** [comment]
Green now.

File: tests/system/docker-compose.yml [hunk 3]
```diff
@@ -93,4 +93,4 @@ services:
       - node-6
       - node-7
     volumes:
-      - "./node/faucet/:/root/faucet/"
+      - "./node/faucet/config.yaml:/root/faucet/config.yaml"
```
Comment (Geod24) [inline tests/system/docker-compose.yml:96]: @kchulj : That's another option!

**Review by Geod24 — APPROVED** [review]

**Comment by Geod24** [comment]
Thanks! Regarding the Faucet bug, it was observed today, and I recommended to @kchulj that he use a similar solution to Agora:
https://github.com/bosagora/agora/commit/4dff89356006021b738fc536a28a5f324c5e5d56

**Pull request merged**